strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/pypi/PypiArtifactController.java

```diff
@@ -0,0 +1,351 @@
+                    repository.getStorage().getId(),
+                    repository.getId(), packageName);
+
+        final Map<String, String> uriVariables = new HashMap<String, String>();
+        uriVariables.put("storageId", repository.getStorage().getId());
+        uriVariables.put("repositoryId", repository.getId());
+        uriVariables.put("packageName", packageName);
+
+        final URI location = ServletUriComponentsBuilder
+                                                        .fromCurrentServletMapping()
+                                                        .path("/storages/{storageId}/{repositoryId}/simple/{packageName}/")
+                                                        .build()
+                                                        .expand(uriVariables)
+                                                        .toUri();
+
+        final HttpHeaders headers = new HttpHeaders();
+        headers.setLocation(location);
+        return ResponseEntity.status(HttpStatus.SEE_OTHER)
+                             .headers(headers)
+                             .body(HttpStatus.SEE_OTHER.getReasonPhrase());
+    }
+
+    @ApiOperation(value = "This Endpoint will be used to retreive pypi artifact/.whl file")
+    @ApiResponses(value = { @ApiResponse(code = HttpURLConnection.HTTP_OK, message = "Success"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_NOT_FOUND, message = "Request Url Not Found"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_INTERNAL_ERROR, message = "An error occurred while executing download request."),
+                            @ApiResponse(code = HttpURLConnection.HTTP_UNAVAILABLE, message = "Service Unavailable.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(path = "{storageId}/{repositoryId}/packages/{artifactName:.+}", method = RequestMethod.GET)
+    public void downloadPackage(@RepositoryMapping Repository repository,
+                                @PathVariable(name = "artifactName") String artifactName,
+                                HttpServletRequest request,
+                                HttpServletResponse response,
+                                @RequestHeader HttpHeaders headers)
+        throws Exception
+    {
+
+        logger.info("Download package request for storageId -> [{}] , repositoryId -> [{}], artifactName -> [{}]",
+                    repository.getStorage().getId(),
+                    repository.getId(), artifactName);
+
+        // String[] artifactNameTokens = artifactName.split("-");
+        // if (artifactNameTokens.length < 5)
+        // {
+        // response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        // return;
+        // }
+
+        // final String packageName =
+        // artifactNameTokens[0].replaceAll("[^A-Za-z0-9 ]", "-");
+        // final String version = artifactNameTokens[1];
```

Comment:
@anki2189 can you please drop this commented part?

Reply:
Sure